examples/todomvc/src/components/Footer.js

```diff
@@ -9,64 +9,40 @@ const FILTER_TITLES = {
   [SHOW_COMPLETED]: 'Completed'
 }
 
-export default class Footer extends Component {
-  static propTypes = {
-    completedCount: PropTypes.number.isRequired,
-    activeCount: PropTypes.number.isRequired,
-    filter: PropTypes.string.isRequired,
-    onClearCompleted: PropTypes.func.isRequired,
-    onShow: PropTypes.func.isRequired
-  }
-
-  renderTodoCount() {
-    const { activeCount } = this.props
-    const itemWord = activeCount === 1 ? 'item' : 'items'
-
-    return (
+const Footer = (props) => {
+  const { activeCount, completedCount, onClearCompleted } = props
+  const itemWord = activeCount === 1 ? 'item' : 'items'
+  return (
+    <footer className="footer">
       <span className="todo-count">
         <strong>{activeCount || 'No'}</strong> {itemWord} left
       </span>
-    )
-  }
-
-  renderFilterLink(filter) {
-    const title = FILTER_TITLES[filter]
-    const { filter: selectedFilter, onShow } = this.props
-
-    return (
-      <a className={classnames({ selected: filter === selectedFilter })}
-         style={{ cursor: 'pointer' }}
-         onClick={() => onShow(filter)}>
-        {title}
-      </a>
-    )
-  }
-
-  renderClearButton() {
-    const { completedCount, onClearCompleted } = this.props
-    if (completedCount > 0) {
-      return (
-        <button className="clear-completed"
-                onClick={onClearCompleted} >
-          Clear completed
-        </button>
-      )
-    }
-  }
+      <ul className="filters">
+        {Object.keys(FILTER_TITLES).map(filter =>
+          <li key={filter}>
+            <FilterLink filter={filter}>
+              {FILTER_TITLES[filter]}
+            </FilterLink>
+          </li>
+        )}
+      </ul>
+      {
+        (completedCount > 0) ?
```

Comment:
How about doing `!!completedCount && <Component ... />` instead of the ternary? Would also get rid of the `null` being returned if the condition is not met.